metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -250,10 +250,20 @@ object MetalsEnrichments
       }
     }
     def isDependencySource(workspace: AbsolutePath): Boolean =
-      workspace.toNIO.getFileSystem == path.toNIO.getFileSystem &&
-        path.toNIO.startsWith(
-          workspace.resolve(Directories.readonly).toNIO
-        )
+      isLocalFileSystem(workspace) &&
+        isInReadonlyDirectory(workspace)
+
+    def isWorkspaceSource(workspace: AbsolutePath): Boolean =
```

Comment:
`path.startsWith(workspace)` should strictly speaking be a condition as well

Reply:
Added!